components/user_settings/display/manage_timezones/manage_timezones.tsx

```diff
@@ -147,9 +164,48 @@ export default class ManageTimezones extends React.PureComponent<Props, State> {
     handleManualTimezone = (e: React.ChangeEvent<HTMLSelectElement>) => {
         this.setState({manualTimezone: e.target.value});
     };
+    handleContainerKeyDown = (e: KeyboardEvent) => {
+        const modalBody = document.querySelector('.modal-body');
+        if (isKeyPressed(e, Constants.KeyCodes.ESCAPE) && this.state.openMenu) {
+            modalBody?.classList.remove('no-scroll');
+            this.setState({openMenu: false});
+            e.stopPropagation();
+        }
+    };
+
+    handleKeyDown = (e: React.KeyboardEvent) => {
+        const modalBody = document.querySelector('.modal-body');
```

Comment:
Please avoid using query selector and use classNames, We’re aiming to have a declarative codebase where possible. Also we have a pattern of identifying model you can take a look at examples in edit_channel_header_modal.jsx

We have close and open actions for modals as well as modal identifiers here
import {openModal, closeModal} from 'actions/views/modals';